### ISSUE TITLE
Column Select (<shift>+mouse-drag) with Rpl All do not work together as expected

### ISSUE THREAD

**Issue by G4OEU** [DistributedProofreaders/guiguts#978]
**Describe the bug**
If you use Column Select to select a block of text then do Rpl All to replace all instances of a character within the selected block, only the character on the bottom line of the block is replaced. The behaviour is present in both 1.4.0 and 1.4.1-test5.

**To Reproduce**
1. Open Guiguts
2. Open 
[bug-example.txt](https://github.com/DistributedProofreaders/guiguts/files/9409552/bug-example.txt)
3. Open the Search->Search & Replace menu 
4. Use column select to select (say) columns 4,5 and 6 in the example file
5. Use Search & Replace->Rpl All to replace the space characters with (say) the letter 'Q' in the text block highlighted by Step 4 
6. Note how only the bottom line of the highlighted text block is changed

**Expected behavior**
I expected that Rpl All would replace the space character throughout the whole highlighted block

**Environment**
* Guiguts version: 1.4.0 and 1.4.1-test5
* OS and version:  Windows 10
* Perl version: v5.32.1
* Perl distribution: 

**Screenshots**

**Additional context**

**Comment by windymilla** [comment]
This is related to #144 which was never fixed.
S&R has never worked on column selections - I tested back to 1.0.25
The code is not intended to handle that case - I'll look to see if I can safely make it do that.

**Comment by windymilla** [comment]
TL;DR: I've looked, and I can't. 

Column selection is stored in the code as a list of selection start and end points (e.g. L1C3, L1C6, L2C3, L2C6, etc), and many things only look at the final start and end points rather than processing the whole list of selection points. Replace All is one of these, so it only makes changes within the last selected bit. ReplAll works by repeatedly doing a search and replace. Unfortunately, the Search routine also has code that looks at the selected ranges, and attempting to call it repeatedly for each selection chunk makes it go badly wrong. I think a substantial change to the way that search and replace are coded would be the way to fix it, but I dare not do that. It might be possible to put a wrapper round the ReplAll code, essentially mimicking selecting the first line of your column select, then doing a ReplAll, then selecting the second line of your column select, doing a ReplAll, etc., but that would be quite a mess. 

Note to anyone attempting this: I think you would also have to convert those selection start/end points into "marks" stored in the text widget, since the indexes currently used might go wrong if you replaced with a different length string than the one you searched for.

So, reluctantly, I think I have to say "no" to this one.

**Comment by G4OEU** [comment]
Thanks for looking at it. It is not something I have tried before. I happened to be replacing text in some tables and selecting columns spanning multiple lines was possible, rather than selecting/replacing line by line. I will just revert to the latter.

**Comment by windymilla** [comment]
Thanks Quentin
